tests/publiccloud/create_aistack_env.pm

```diff
@@ -0,0 +1,247 @@
+                if ($logs =~ /ERROR|FAILURE|Exception|Failed/) {
+                    record_info("$pod failed due to error in log: $logs \n ");
+                    next POD_LOOP;
+                }    # if log
+                sleep $sleep_interval;
+            }    # if status
+        }    # while loop
+        record_info("$pod is not running after $max_retries ");
+    }    #pod loop
+
+    assert_script_run("kubectl get all --namespace $namespace");
+    record_info("Logs for the pods which is not in running or pending state");
+    foreach my $pod (@out) {
+        my $status = script_output("kubectl get pod $pod -n $namespace -o=jsonpath='{.status.phase}'", proceed_on_failure => 1);
+        if ($status !~ /^(Running|Pending|Completed)$/) {
+            my $logs = script_output("kubectl logs $pod -n $namespace", proceed_on_failure => 1);
+            record_info("$pod is in $status state. Logs:\n$logs\n");
+        }
+    }    # pod loop
+}
+
+sub test_openwebui_service {
+    my ($instance, $namespace) = @_;
+    my $sr_name = 'open-webui';
+    my $host_name = 'suse-ollama-webui';
+    #my $host_name = get_var('OPENWEBUI_HOSTNAME');
+    record_info('OpenWebUI service');
+
+    # After successfull installation, Get open-webUI ipaddress and add in /etc/host and verify connectivity
+    assert_script_run("kubectl get ingress --namespace $namespace -o json");
+    my $ipaddr = script_output("kubectl get ingress -n $namespace -o jsonpath='{.items[0].status.loadBalancer.ingress[0].ip}'");
+    assert_script_run("echo \"$ipaddr $host_name\" | sudo tee -a /etc/hosts > /dev/null");
+    set_var('OPENWEBUI_IP', "$ipaddr");
+    record_info("Added $ipaddr to /etc/hosts with hostname $host_name");
+    my $curl_cmd = "curl -v http://$host_name";
+    my $curl_result = script_run($curl_cmd);
+    if ($curl_result == 0) {
+        record_info("Successfully connected to the open-webui service at $curl_cmd \n");
+    } else {
+        die "Unable to connect to the open-webui service at $curl_cmd\n";
+    }
+}
+
+
+sub install_nvidia_drivers {
+    my ($instance, $values_url, $file_name) = @_;
+    my $easyinstall_url = 'https://gitlab.nue.suse.com/cloud-solutions-sys-eng/nvidia-drivers-easy-install/-/blob/main/nvidia_easy_install.sh';
+    record_info('Install nvidia drivers');
+    my $driver_version = '550.54.14';
+    #my $gpu_op_url = get_var('GPU_OPERATOR');
+    my $gpu_op_url = 'nvidia https://nvidia.github.io/gpu-operator';
+
+#    my $driver_version=script_output("zypper se -s nvidia-open-driver | grep nvidia-open-driver- | sed 's/.* package | //g' | sed 's/\s.*//g' | sort | head -n 1 | sed 's/[-_].*//g'");
+    script_run("sudo zypper ar https://download.nvidia.com/suse/sle15sp6/ nvidia-sle15sp6-main");
+    script_run("sudo zypper --gpg-auto-import-keys refresh");
+    script_run("sudo zypper install -y --auto-agree-with-licenses nvidia-open-driver-G06-signed-kmp=$driver_version nvidia-compute-utils-G06=$driver_version");
```

Comment:
See how this is handled at publiccloud/nvidia, likely there's a way to reuse part of the code

https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/c1b7752a78aaa4486296e42313831ad72e9efe64/tests/publiccloud/nvidia.pm#L32

Reply:
I cannot reuse this code.
`  trup_call("pkg install -y --auto-agree-with-licenses nvidia-open-driver-G06-signed-kmp=$driver_version nvidia-compute-utils-G06=$driver_version");

`